File: apimsdc/MsdcQUsgAAccess.cpp
/**
  * \file MsdcQUsgAAccess.cpp
  * API code for table TblMsdcQUsgAAccess (implementation)
  * \author Alexander Wirthmueller
  * \date created: 29 Aug 2018
  * \date modified: 29 Aug 2018
  */

#ifdef _WIN32
	#include "stdafx.h"
#endif

#include "MsdcQUsgAAccess.h"

/******************************************************************************
 class MsdcQUsgAAccess
 ******************************************************************************/

MsdcQUsgAAccess::MsdcQUsgAAccess(
			const uint jnum
			, const string srefX1IxMsdcVFeatgroup
			, const string titX1IxMsdcVFeatgroup
			, const string x2FeaSrefUix
			, const string srefsIxMsdcWAccess
			, const string titsIxMsdcWAccess
		) {
	this->jnum = jnum;
	this->srefX1IxMsdcVFeatgroup = srefX1IxMsdcVFeatgroup;
	this->titX1IxMsdcVFeatgroup = titX1IxMsdcVFeatgroup;
	this->x2FeaSrefUix = x2FeaSrefUix;
	this->srefsIxMsdcWAccess = srefsIxMsdcWAccess;
	this->titsIxMsdcWAccess = titsIxMsdcWAccess;
};

bool MsdcQUsgAAccess::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "MsdcQUsgAAccess");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "srefX1IxMsdcVFeatgroup", "feg", srefX1IxMsdcVFeatgroup);
		extractStringUclc(docctx, basexpath, "titX1IxMsdcVFeatgroup", "feg2", titX1IxMsdcVFeatgroup);
		extractStringUclc(docctx, basexpath, "x2FeaSrefUix", "fea", x2FeaSrefUix);
		extractStringUclc(docctx, basexpath, "srefsIxMsdcWAccess", "acc", srefsIxMsdcWAccess);
		extractStringUclc(docctx, basexpath, "titsIxMsdcWAccess", "acc2", titsIxMsdcWAccess);
	};

	return basefound;
};

/******************************************************************************
 class ListMsdcQUsgAAccess
 ******************************************************************************/

ListMsdcQUsgAAccess::ListMsdcQUsgAAccess() {
};

ListMsdcQUsgAAccess::ListMsdcQUsgAAccess(
			const ListMsdcQUsgAAccess& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i=0;i<nodes.size();i++) nodes[i] = new MsdcQUsgAAccess(*(src.nodes[i]));
};

ListMsdcQUsgAAccess& ListMsdcQUsgAAccess::operator=(
			const ListMsdcQUsgAAccess& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i=0;i<nodes.size();i++) nodes[i] = new MsdcQUsgAAccess(*(src.nodes[i]));

	return *this;
};

ListMsdcQUsgAAccess::~ListMsdcQUsgAAccess() {
	clear();
};

void ListMsdcQUsgAAccess::clear() {
	for (unsigned int i=0;i<nodes.size();i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListMsdcQUsgAAccess::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	MsdcQUsgAAccess* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListMsdcQUsgAAccess");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "MsdcQUsgAAccess", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i=0;i<ics.size();i++) {
		rec = new MsdcQUsgAAccess(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/MsdcQUsgAAccess[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};

File: japimsdc/MsdcQUsgAAccess.java
/**
  * \file MsdcQUsgAAccess.h
  * Java API code for record of table TblMsdcQUsgAAccess
  * \author Alexander Wirthmueller
  * \date created: 4 Oct 2018
  * \date modified: 4 Oct 2018
  */

package apimsdc;

import org.w3c.dom.*;
import sbecore.*;

public class MsdcQUsgAAccess {
	
	public MsdcQUsgAAccess(
				int jnum
				, String srefX1IxMsdcVFeatgroup
				, String titX1IxMsdcVFeatgroup
				, String x2FeaSrefUix
				, String srefsIxMsdcWAccess
				, String titsIxMsdcWAccess
			) {
		this.jnum = jnum;
		this.srefX1IxMsdcVFeatgroup = srefX1IxMsdcVFeatgroup;
		this.titX1IxMsdcVFeatgroup = titX1IxMsdcVFeatgroup;
		this.x2FeaSrefUix = x2FeaSrefUix;
		this.srefsIxMsdcWAccess = srefsIxMsdcWAccess;
		this.titsIxMsdcWAccess = titsIxMsdcWAccess;
	};

	public int jnum;
	public String srefX1IxMsdcVFeatgroup;
	public String titX1IxMsdcVFeatgroup;
	public String x2FeaSrefUix;
	public String srefsIxMsdcWAccess;
	public String titsIxMsdcWAccess;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "MsdcQUsgAAccess");

		if (Xmlio.checkXPath(doc, basexpath)) {
			srefX1IxMsdcVFeatgroup = Xmlio.extractStringUclc(doc, basexpath, "srefX1IxMsdcVFeatgroup", "feg", null, 0);
			titX1IxMsdcVFeatgroup = Xmlio.extractStringUclc(doc, basexpath, "titX1IxMsdcVFeatgroup", "feg2", null, 0);
			x2FeaSrefUix = Xmlio.extractStringUclc(doc, basexpath, "x2FeaSrefUix", "fea", null, 0);
			srefsIxMsdcWAccess = Xmlio.extractStringUclc(doc, basexpath, "srefsIxMsdcWAccess", "acc", null, 0);
			titsIxMsdcWAccess = Xmlio.extractStringUclc(doc, basexpath, "titsIxMsdcWAccess", "acc2", null, 0);
			
			return true;
		};

		return false;
	};
};

